spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/InferShape.scala

```diff
@@ -71,7 +64,7 @@ trait InferShape {
    */
   private[bigdl] def build(inputShape: Shape): Shape = {
```

Comment:
what if it is already built?

Reply:
Or what if inputShape is different from _inputShapeValue?

Reply:
Or what if inputShape is different from the one user specified for Torch-style layers?

Reply:
`ensureNotShared ` added to ensure a layer can only be built once.
`checkWithCurrentInputShape ` added to ensure the inputshape from "graph" is the same with the user specified one.

Reply:
Would check `isBuilt()` to ensure a KerasLayer being built only once.

Reply:
What if a Model is added to a Sequential? checkWithCurrentInputShape is only used in KerasLayer

Reply:
`Model` would be inherited from `KerasLayer` same as `KSequential`